scripts/is-binary-up-to-date.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BASE_URL="https://binaries.soliditylang.org/bin"
+REPO_ROOT="$(dirname "$0")/.."
+LIST_FILE=$(mktemp -t solc-bin-list-XXXXXX.json)
+
+function fail() {
+    echo -e "ERROR: $*" >&2
+    exit 1
+}
+
+function check_release_version() {
+    local current_version="$1"
+
+    curl --silent --fail "${BASE_URL}/list.json" -o "$LIST_FILE"
+    [[ ! -f $LIST_FILE ]] && fail "Download of release list failed:\n    [url]: ${BASE_URL}/list.json"
+
+    # Retrieve the latest released version
+    latest_version_short=$(jq --raw-output ".latestRelease" "$LIST_FILE")
+    latest_version_long=$(jq --raw-output ".releases | .[\"${latest_version_short}\"]" "$LIST_FILE" | sed --regexp-extended --quiet 's/^soljson-v(.*).js$/\1/p')
+
+    # Check if current version is the latest release
+    if [[ "$current_version" != "$latest_version_long" ]]; then
```

Comment:
As a general rule, inside `[[ ]]` you only need the quotes on the right side. And that's only in cases where the stuff on the right side could be mistakenly interpreted as a regex (quoting it disables that).
```suggestion
    if [[ $current_version != "$latest_version_long" ]]; then
```